src/scheduler/prime.c

```diff
@@ -228,6 +228,34 @@ is_holiday(long date)
 	return 1;
 }
 
+/**
+ * @brief	Set conf.prime values to reflect "ALL PRIME" before we start parsing
+ * 			the holidays file
+ *
+ * @param	void
+ *
+ * @return void
+ */
+static void
+handle_missing_prime_info(void)
```

Comment:
This should happen in init_config()

Reply:
I wanted to do the same thing, but, if somebody adds "all all" as prime and non prime start in holidays file, then we need to be able to catch that in parse_holidays and set all prime-time, not all non-prime time. We don't have a good way of catching that if we set the bits to "all prime" inside init_config() itself as we won't be able to tell inside load_day() whether the bits are all prime because the holidays file has "all" entry for prime, or if it is what was set by init_config(). So, we could refactor the function if you really hate this, but otherwise i can't put it inside init_config().

Reply:
I don't understand.  You call handle_missing_prime_info() right at the top of parse_holidays().  There should be no difference between the same code being called in init_config() or in parse_holidays()

Reply:
No, I call it at the end after parsing the holidays file